File: dotnet/cafdemalihapa/Applications/DirectoryManagement/DirectoryOpenStrategies/LinuxDirectoryOpenStrategy.cs
using System.Runtime.InteropServices;
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Applications.DirectoryManagement.DirectoryOpenStrategies
{
    public class LinuxDirectoryOpenStrategy(ILogger<LinuxDirectoryOpenStrategy> logger)
        : DirectoryOpenStrategy(logger)
    {
        protected override string FileName => "xdg-open";

        public override bool CanHandle() => RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/DirectoryManagement/DirectoryOpenStrategies/LinuxDirectoryOpenStrategy.fs
namespace cafdemwimu.console.Applications.DirectoryManagement.DirectoryOpenStrategies

open System.Runtime.InteropServices
open Microsoft.Extensions.Logging

type LinuxDirectoryOpenStrategy(logger: ILogger<LinuxDirectoryOpenStrategy>) =
    inherit DirectoryOpenStrategy(logger)

    override _.FileName = "xdg-open"

    override _.CanHandle() = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
